src/ui/configuration/ApmSoftwareConfig.cc

```diff
@@ -327,17 +336,20 @@ void ApmSoftwareConfig::activeUASSet(UASInterface *uas)
 #else
     QString appDataDir = getenv("HOME");
 #endif
-    QDir autopilotdir(QGC::shareDirectory() + "/files/" + uas->getAutopilotTypeName().toLower());
-    m_apmPdefFilename = autopilotdir.absolutePath() + "/" + vehicle_pdef_filename;
-
-    if (!QFile::exists(m_apmPdefFilename))
+    QDir autopilotdir(QGC::appDataDirectory());
+    if(autopilotdir.cd(s_xmlSubFolder))
+    {
+        m_apmPdefFilename = autopilotdir.filePath(vehicle_pdef_filename);
+    }
+    else
     {
-        m_apmPdefFilename = QDir(appDataDir + "/apmplanner2").filePath("apm.pdef.xml");
+        m_apmPdefFilename = QDir(appDataDir + "/apmplanner2").filePath("apm.pdef.xml"); // Fall back
     }
 
     QFile xmlfile(m_apmPdefFilename);
-    if (xmlfile.exists() && !xmlfile.open(QIODevice::ReadOnly))
+    if (!xmlfile.exists() || (xmlfile.exists() && !xmlfile.open(QIODevice::ReadOnly)))
```

Comment:
You don't need the `xmlfile.exists() &&` part, if you reach that part of the condition you already know the file exists.

Reply:
Yes - you are right - I will fix it.
Thank you :+1: